src/main/java/com/google/cloud/tools/managedcloudsdk/process/ByteHandler.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.process;
+
+/**
+ * Handle bytes from a stream consumer.
+ *
+ * @param <T> use {@code Void} if you don't want to store the result
+ */
+public interface ByteHandler<T> {
+  void bytes(byte[] bytes, int length);
```

Comment:
If length is > bytes.length what happens?

Reply:
Presumably the caller will never do that as this is just a passthrough.